glean-core/ffi/src/upload.rs

```diff
@@ -147,9 +145,13 @@ impl Drop for FfiPingUploadTask {
             unsafe {
                 glean_str_free(*document_id);
                 glean_str_free(*path);
-                glean_str_free(*body);
                 glean_str_free(*headers);
             }
+            // Unfortunately, we cannot directly call `body.destroy();` as
+            // we're behind a mutable reference, so we have to manually take the
+            // ownership and drop.
```

Comment:
nit: maybe mention that `ByteBuffer::new_with_size(0)` does not allocate, so we're not leaking memory.